MinnPost/object-sync-for-salesforce
# 200 prematch against meta fields

## What does this PR do?

Fixes issue #200 in which prematch against a meta field would fail. This addresses the issue by:

1. adding a `pull_to_drafts` option to fieldmaps to ensure that users want to update/match against draft posts, if applicable for the object type.
2. adding a `match` parameter to the `fields` row in the database, so we can ensure we have the correct match method for metadata cases.
3. use WP_Query, WP_User_Query, WP_Term_Query, WP_Comment_Query for matching records by a metadata field.

## How do I test this PR?

- use a fieldmap that matches against a value that is stored in a meta table
- I've also been testing by using a fieldmap that matches against a value that is stored in the not-meta table to make sure this continues to behave as it previously did.
